### ISSUE TITLE
Weight decay scaling?

### ISSUE THREAD

**Issue by RoyHJ** [NVIDIA/apex#617]
Is weight decay scaled by AMP?
(http://on-demand.gputechconf.com/gtc/2017/presentation/s7218-training-with-mixed-precision-boris-ginsburg.pdf page 15)

**Comment by mcarilli** [comment]
Regardless of how weight decay is implemented (whether by adding the L2 norm of the weights as a penalty term to the loss, or by requesting that the optimizer instance handle it) weight decay should work properly in the presence of loss scaling with Amp as implemented here.

That talk is out of date, [this one](https://github.com/mcarilli/mixed_precision_references/blob/master/GTC_2019/Session_9998_carilli.pdf) is a better reference that is specific to Pytorch and Apex.  You should be aware, my main task right now is moving Amp into Pytorch core, see https://github.com/pytorch/pytorch/issues/25081.
